zen/screen-renderer.c

```diff
@@ -177,6 +177,71 @@ zn_screen_renderer_render_view_popups(struct zn_screen *screen,
   }
 }
 
+static void
+zn_screen_renderer_render_decoration(struct zn_screen *screen,
+    struct zn_view *view, struct wlr_renderer *renderer,
+    pixman_region32_t *screen_damage)
+{
+  struct zn_output *output = screen->output;
+  struct wlr_box transformed_box;
+  pixman_region32_t render_damage;
+  pixman_box32_t *rects;
+  float matrix[9];
+  int rect_count;
+  struct wlr_fbox fbox;
+
+  zn_view_get_decoration_fbox(view, &fbox);
+
+  zn_output_box_effective_to_transformed_coords(
+      output, &fbox, &transformed_box);
+
+  pixman_region32_init(&render_damage);
+  pixman_region32_union_rect(&render_damage, &render_damage, transformed_box.x,
+      transformed_box.y, transformed_box.width, transformed_box.height);
+
+  pixman_region32_intersect(&render_damage, &render_damage, screen_damage);
```

Comment:
I think it's better to avoid having the same variable name among function scope and inner block scope.
(`fbox` and `transformed_box`)

How about adding an additional inner scope?
```suggestion
{
  struct wlr_fbox fbox;
  struct wlr_box transformed_box;
  zn_view_get_decoration_fbox(view, &fbox);

  zn_output_box_effective_to_transformed_coords(
      output, &fbox, &transformed_box);

  pixman_region32_init(&render_damage);
  pixman_region32_union_rect(&render_damage, &render_damage, transformed_box.x,
      transformed_box.y, transformed_box.width, transformed_box.height);

  pixman_region32_intersect(&render_damage, &render_damage, screen_damage);
}
```

Reply:
> avoid having the same variable name

That's true, so I will use different name in that scope